pkg/sdk/connections_def.go

```diff
@@ -0,0 +1,121 @@
+package sdk
+
+//go:generate go run ./poc/main.go
+
+import (
+	g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+)
+
+var ConnectionDef = g.NewInterface(
+	"Connections",
+	"Connection",
+	g.KindOfT[AccountObjectIdentifier](),
+).CreateOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/create-connection",
+	g.NewQueryStruct("CreateConnection").
+		Create().
+		SQL("CONNECTION").
+		IfNotExists().
+		Name().
+		OptionalComment().
+		WithValidation(g.ValidIdentifier, "name"),
+).CustomOperation(
+	"CreateReplicated",
```

Comment:
nit: CreateAsReplicaOf

Reply:
Or CreateSecondary to be consistent with other objects that can be replicated (e.g. databases)